wire-ios/Wire-iOS/Sources/Helpers/syncengine/Conversation+Participants.swift

```diff
@@ -39,18 +39,34 @@ extension ZMConversation {
 
     func addOrShowError(participants: [UserType]) {
         guard let session = ZMUserSession.shared(),
-                session.networkState != .offline else {
+              session.networkState != .offline else {
             self.showAlertForAdding(for: NetworkError.offline)
             return
         }
 
         addParticipants(participants) { (result) in
-            switch result {
-            case .failure(let error):
-                self.showAlertForAdding(for: error)
+            guard case .failure(let error) = result else {
+                return
+            }
+
+            switch error {
+            case .unreachableUsers(let users):
+                let withoutUnreachable = participants.filter { user in
+                    guard let userId = user.remoteIdentifier else {
+                        return false
+                    }
+                    return !userId.isOne(of: users.map(\.remoteIdentifier))
```

Comment:
This could be expensive, for each participant we need to compare it with each user. I wonder if we can use set operations to compute this more efficiently. For instance:

```swift
let reachableUsers = Set(participants).subtracting(Set(users))
```

Reply:
done